store/store.go

```diff
@@ -583,46 +699,59 @@ func (c *MongoStoreClient) GetLoopableMedtronicDirectUploadIdsAfter(userID strin
 		"deviceModel": bson.M{"$in": []string{"523", "523K", "554", "723", "723K", "754"}},
 	}
 
-	legacyObjects := []struct {
-		UploadID string `bson:"uploadId"`
-	}{}
-
-	objects := []struct {
+	type Object struct {
 		UploadID string `bson:"uploadId"`
-	}{}
+	}
+	var legacyObjects []Object
+	var objects []Object
 
 	cursor, err := dataCollection(c).Find(c.context, query, opts)
 	if err != nil {
 		return nil, err
 	}
 
-	legacyCursor, err := dataCollection(c).Find(c.context, legacyQuery, legacyOpts)
-	if err != nil {
-		return nil, err
+	collections := []*mongo.Collection{
+		dataCollection(c),
+		dataSetsCollection(c),
 	}
+	for _, collection := range collections {
+		legacyCursor, err := collection.Find(c.context, legacyQuery, legacyOpts)
```

Comment:
`legacyQuery` fetches records which have a string `time` attribute. I think at this point all records are converted and use native date time type. @Roukoswarf to confirm.

I think you should either remove the legacy query completely or do 4 queries in total - both collections with both time field types.

Reply:
I would recommend rebasing on #105 as it removes all ambiguity in that case, and corrects all unit tests to remove all legacy dates.